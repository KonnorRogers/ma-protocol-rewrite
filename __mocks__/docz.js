const docz = jest.genMockFromModule("docz")

// name, route, id
const doc1 = {
  id: 0,
  route: "/",
  name: "firstDoc",
}
const doc2 = {
  id: 1,
  route: "/second-doc",
  name: "secondDoc",
}
const doc3 = {
  id: 2,
  route: "/second-doc",
  name: "secondDoc",
}

const allDocz = [doc1, doc2, doc3]

let currentDoc = doc1

function setCurrentDoc(id) {
  currentDoc = allDocz[id]
}

function useCurrentDoc() {
  return currentDoc
}

function useDocs() {
  return allDocz
}

docz.setCurrentDoc = setCurrentDoc
docz.useCurrentDoc = useCurrentDoc
docz.useDocs = useDocs

module.exports = docz
