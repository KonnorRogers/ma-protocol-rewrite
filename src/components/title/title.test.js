import React from "react"
import { render } from "@testing-library/react"
import Title from "./title.js"

// Mocked hook
import { setCurrentDoc } from "docz"

jest.mock("docz")

test("Renders without error when given a title", () => {
  setCurrentDoc(1)
  const { asFragment } = render(<Title title="Title Test" />)

  expect(asFragment).toMatchSnapshot()
})

test("Renders without error when not given a title for 1st doc", () => {
  setCurrentDoc(0)
  const { asFragment } = render(<Title />)
  expect(asFragment).toMatchSnapshot()
})

test("Renders without error when not given a title for 2nd doc", () => {
  setCurrentDoc(1)
  const { asFragment } = render(<Title />)
  expect(asFragment).toMatchSnapshot()
})

test("Renders without error when not given a title for 3rd doc", () => {
  setCurrentDoc(2)
  const { asFragment } = render(<Title />)
  expect(asFragment).toMatchSnapshot()
})
