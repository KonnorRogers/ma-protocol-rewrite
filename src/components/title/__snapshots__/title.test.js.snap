// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without error when given a title 1`] = `[Function]`;

exports[`Renders without error when not given a title for 1st doc 1`] = `[Function]`;

exports[`Renders without error when not given a title for 2nd doc 1`] = `[Function]`;

exports[`Renders without error when not given a title for 3rd doc 1`] = `[Function]`;
